src/ApiService/ApiService/onefuzzlib/ScalesetOperations.cs

```diff
@@ -693,33 +695,32 @@ where x.State.ReadyForReset()
             return;
         }
 
-        HashSet<Guid> machineIds = new();
+        var nodesToDelete = new List<Node>();
         foreach (var node in nodes) {
             if (node.DebugKeepNode) {
-                _log.Warning($"not deleting manually overriden node {node.MachineId:Tag:MachineId} in scaleset {scaleset.ScalesetId:Tag:ScalesetId}");
+                _log.Warning($"not deleting manually overridden node {node.MachineId:Tag:MachineId} in scaleset {scaleset.ScalesetId:Tag:ScalesetId}");
             } else {
-                _ = machineIds.Add(node.MachineId);
+                nodesToDelete.Add(node);
             }
         }
 
         switch (disposalStrategy) {
             case NodeDisposalStrategy.Decommission:
                 _log.Info($"decommissioning nodes");
-                await Async.Task.WhenAll(nodes
-                    .Where(node => machineIds.Contains(node.MachineId))
+                await Async.Task.WhenAll(nodesToDelete
                     .Select(async node => {
                         await _context.NodeOperations.ReleaseScaleInProtection(node).IgnoreResult();
                     }));
                 return;
 
             case NodeDisposalStrategy.ScaleIn:
-                _log.Info($"deleting nodes {scaleset.ScalesetId:Tag:ScalesetId} {string.Join(", ", machineIds):Tag:MachineIds}");
-                await _context.VmssOperations.DeleteNodes(scaleset.ScalesetId, machineIds);
-                await Async.Task.WhenAll(nodes
-                    .Where(node => machineIds.Contains(node.MachineId))
+                _log.Info($"deleting nodes {scaleset.ScalesetId:Tag:ScalesetId} {string.Join(", ", nodesToDelete.Select(n => n.MachineId)):Tag:MachineIds}");
+                await _context.VmssOperations.DeleteNodes(scaleset.ScalesetId, nodesToDelete);
+                await Async.Task.WhenAll(nodesToDelete
                     .Select(async node => {
                         await _context.NodeOperations.Delete(node);
-                        await _context.NodeOperations.ReleaseScaleInProtection(node).IgnoreResult();
+                        // REMOVE?: don't need to release scale-in protection if we have deleted the node
```

Comment:
this removal to be checked

Reply:
@tevoinea I think you’re the most familiar with this code/sequence

Reply:
I think this is also the cause of the "not an active Instance ID" exceptions we are seeing

Reply:
I think you're right, we could probably skip releasing scale in protection here.